extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -96,20 +121,44 @@ export class ScrollManager {
 
   /**
    * @param {!../../../src/service/viewport/viewport-interface.ViewportChangedEventDef} e
+   * @return {!Promise}
    * @private
    */
   onScroll_(e) {
-    /** {./scroll-manager.ScrollEventDef} */
-    const scrollEvent = {
-      top: e.top,
-      left: e.left,
-      width: e.width,
-      height: e.height,
-      scrollWidth: this.viewport_.getScrollWidth(),
-      scrollHeight: this.viewport_.getScrollHeight(),
-    };
-    // Fire all of our children scroll observables
-    this.scrollObservable_.fire(scrollEvent);
+    const firstMeasure = this.initialRootLayoutRect_ === null;
+    return this.getInitRootElementRect_().then(initialRootLayoutRect => {
+      const {
+        height: initialScrollHeight,
+        width: initialScrollWidth,
+      } = initialRootLayoutRect;
+
+      return (firstMeasure
+        ? this.getInitRootElementRect_()
+        : this.measureRootElement_()
+      ).then(layoutRect => {
+        const {
+          top: scrollTop,
+          left: scrollLeft,
+          width: scrollWidth,
+          height: scrollHeight,
+        } = layoutRect;
+        /** {./scroll-manager.ScrollEventDef} */
+        const scrollEvent = {
+          top: e.top - scrollTop,
```

Comment:
> If I understand correctly, it does not, the shadow docs will have a viewport service for their own ampdoc instance ( AmpDocShadow as opposed to AmpDocSingle) so viewport.getScrollTop will yield the distance between top of the page and the position of the shadow doc.

I'm a bit lost : ) If that's the case, `e.top` is the distance of the element to the shadow doc top. Do we still need to minus the `scrollTop` here?

Reply:
Yes :)) So that an element placed at the top of the shadow doc would be considered at 0% of the top (where top refers to the top of the embedded document, not the entire page)